enhancements/installer/aws-custom-edge-machineset-local-zones.md

```diff
@@ -0,0 +1,677 @@
+- 3 compute worker nodes on the private subnets in the default Availability Zones (parent region)
+- 1 compute edge node on the public subnet in Local Zone
+
+### Workflow Description
+
+TBD
+
+#### Variation [optional]
+
+<!-- If the cluster creator uses a standing desk, in step 1 above they can
+stand instead of sitting down.  
+
+See
+https://github.com/openshift/enhancements/blob/master/enhancements/workload-partitioning/management-workload-partitioning.md#high-level-end-to-end-workflow
+and https://github.com/openshift/enhancements/blob/master/enhancements/agent-installer/automated-workflow-for-agent-based-installer.md for more detailed examples. -->
+
+TBD
+
+### API Extensions
+
+Not applied. No new APIs will be added.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Public subnet and outbound traffic
+
+It is a best practice to place the worker nodes in private subnets, although the
+AWS provides limited services in the Local Zones requiring a more complex setup to do so.
+
+Those are some options to create subnets on Local Zones:
+
+1) Create a subnet and associate it to a route table with the internet gateway as default.
+2) Create a private subnet, associating it to a route table from the zone in the
+  parent region. The cost will be twice as the traffic should cross the zone to
+  access the internet through a Nat Gateway, which also has fees to be applied.
+3) Create a private subnet, and provide access to the node through a proxy. Same costs
+  tradeoffs of option 2.
+
+Option 1 is covered by this enhancement to create the MachineSets (phase 1)
+and to automate by the installer (phase 2).
+
+The public subnet means the node will receive a Public IP address and the Security
+Group must not expose any ports to the internet. This must be the default behavior,
+and the security group for the `worker` pool will be used on the `edge` pool.
+
+> TODO: check if the current deployment is not exposing any port to the `0.0.0.0/0` on `worker` SG
+
+#### User-workload deployments
+
+Application developers must set the tolerations for the application running for each location.
+
+The pod spec example below describes how to deploy one application on the node on the zone group `ZONE_GROUP`:
+
+```yaml
+      nodeSelector:
+        zone_group: ZONE_GROUP
+      tolerations:
+      - key: "node-role.kubernetes.io/edge"
+        operator: "Equal"
+        value: ""
+        effect: "NoSchedule"
+```
+
+<!--
+> TODO create one example of deploying the application on the same ZONE_GROUP, spreading it into different zones.
+
+> Currently only Los Angeles (LAX) location has more than one zone[1];
+
+The example below shows how to deploy one application with two replicas distributed
+between two Local Zones[1] in the same zone group[2].
+
+[1] Local Zones available in Los Angeles `us-west-2-lax-1a` and `us-west-2-lax-1b`
+[2] Local Zone group of Los Angeles `us-west-2-lax-1` -->
+
+
+#### User-workload ingress traffic
```

Comment:
This section might have to be in the docs. ;)

Reply:
[x-ref this thread](https://github.com/openshift/installer/pull/6371/files#diff-affa4217d0878d1b05bbd3a78a4e9dbbb537dfeb5fb45674aba3646655201fe0R406-R420). Thanks! =)